### ISSUE TITLE
Consider making Path a struct, but using a Builder pattern

### ISSUE THREAD

**Issue by jsbean** [dn-m/PathTools#51]
`Path` is implemented as a `class` to enable a fluent interface for the build process.

Instead, we could implement `Path` as a value type `struct`, with a `Path.Builder` `class`. Use `Path` only as a storage of `PathElements`, and not also an interface for creating them.

```Swift
extension Path {
    final class Builder {
        var elements: [PathElement] = []
        // move all of the `move(to:)`, `addLine(to:)` methods from `Path` to `Builder`.
        func build() -> Path
    }
}
```

In action:

```Swift
let builder = Path.Builder()
    .move(to: p0)
    .addLine(to: p1)
    .addCurve(to: p2, control1: p3, control2: p4)
    .close()
let path = builder.build()
```

**Comment by jsbean** [comment]
Consider making extra interface for the first step, allowing only `move(to:)` using a step build pattern.

**Comment by jsbean** [comment]
Closed by ddf18c44502f0d65804af0ae441fa2cd650289c0.
